qa/tasks/scripts/munge_policy.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+# munge_policy.sh
+#
+# Adds or removes roles from policy.cfg
+#
+
+set -ex
+
+OPERATION=$1 # add_node_as_role / remove_storage_only_node
+NODE_FQDN="${2}" # target192168000063.teuthology
+ROLE="${3}" # storage
+POLICY_CFG="${4}"
+
+function number_of_hosts_in_ceph_osd_tree {
+    ceph osd tree -f json-pretty | jq '[.nodes[] | select(.type == "host")] | length'
+}
+
+function number_of_osds_in_ceph_osd_tree {
+    ceph osd tree -f json-pretty | jq '[.nodes[] | select(.type == "osd")] | length'
+}
+
+function run_stage {
+    local stage_num=$1
+
+    DEV_ENV=true timeout 60m deepsea \
+        --log-file=/var/log/salt/deepsea.log \
+        --log-level=debug \
+        salt-run state.orch ceph.stage.$stage_num \
+        --simple-output
+}
+
+function add_node_as_role {
+    echo "role-${ROLE}/cluster/${NODE_FQDN}.sls" >> ${POLICY_CFG}
+    echo "role-${ROLE} added for: ${NODE_FQDN}"
+}
+
+function remove_node {
+    STORAGE_NODES_BEFORE=$(number_of_hosts_in_ceph_osd_tree)
+    OSDS_BEFORE=$(number_of_osds_in_ceph_osd_tree)
+    test "$STORAGE_NODES_BEFORE"
+    test "$OSDS_BEFORE"
+    test "$STORAGE_NODES_BEFORE" -gt 1
+    test "$OSDS_BEFORE" -gt 0
+
+    sed -i -e "/${NODE_FQDN}/d" ${POLICY_CFG}
+    cat ${POLICY_CFG}
+    run_stage 2
+    run_stage 5
+    ceph -s
+
+    STORAGE_NODES_AFTER=$(number_of_hosts_in_ceph_osd_tree)
+    OSDS_AFTER=$(number_of_osds_in_ceph_osd_tree)
+    test "$STORAGE_NODES_BEFORE"
+    test "$OSDS_BEFORE"
+    test "$STORAGE_NODES_AFTER" -eq "$((STORAGE_NODES_BEFORE - 1))"
```

Comment:
What will happen if there where no nodes before, I think fail, is this expected behavior?